### DISCUSSION TITLE
SOCIAL_AUTH environment variables are not loaded on configuration.py

### DISCUSSION THREAD

**Discussion by appleoddity** [netbox-community/netbox-docker#1052] [Getting Help] [answered]
I'm trying to setup Azure AD integration, as described here: https://docs.netbox.dev/en/stable/administration/authentication/microsoft-azure-ad/

However, being this is a containerized instance, I chose to put the variables in the `env/netbox.env` file. The `REMOTE_AUTH` variables are pulled in and so the login with your IDP option is enabled on the login screen. However, the login fails, because it appears the `SOCIAL_AUTH` values are not pulled in, such as `SOCIAL_AUTH_AZUREAD_OAUTH2_KEY`.

Reading the netbox-docker documentation seems to suggest that these configuration files should not be modified and that the `docker-compose.override.yml` or `env/*` files should be modified.

Is this a bug that should be reported? Is there a reason these environment variables are not included in `configurations.py`? What is the correct way to add the SOCIAL_AUTH variables I need?

**Comment by tobiasge** [comment, marked as answer]
Not all configuration options can be set  with ENV variables. But the container will read all `.py` files from `/etc/netbox/config/`.
If you are using the default docker-compose files you can simply put a new Python file into the `configuration` directory which is mounted into the container.

**Comment by appleoddity** [comment]
Ok. Yes, I added another .py file and the settings were included. Thank you.
